experimental/dds/tree2/src/test/snapshots/testTrees.ts

```diff
@@ -82,26 +96,171 @@ function generateTreeRecursively(
 	}
 }
 
-// TODO:
-// The implementation and usage of these implies the edit history of these trees is relevant, but its not exactly clear how.
-// More documentation on what kind of coverage over possible tree histories and contents this is supposed to provide is needed here.
-// Depending on the above, maybe these tests (or at least some of the cases) should probably be changed to use branches and not full trees to avoid depending on a Fluid runtime.
-// Currently these tests all replicate pre-attachment states.
-// Coverage for other states should be added (including cases with collaboration).
-export function generateTestTrees(): { name: string; tree: () => ISharedTree }[] {
-	return [
+export function generateTestTrees() {
+	const testTrees: {
+		only?: boolean;
+		skip?: boolean;
+		name: string;
+		tree: (takeSnapshot: (tree: ISharedTree, name: string) => Promise<void>) => Promise<void>;
```

Comment:
```suggestion
		runScenario: (takeSnapshot: (tree: ISharedTree, name: string) => Promise<void>) => Promise<void>;
```

now that this is no longer just producing 'a tree' seems fit to rename it.

Also--I'd update the usages of `validate` (old name) to `takeSnapshot` as well.